src/python_testing/conformance_support.py

```diff
@@ -393,3 +395,59 @@ def parse_callable_from_xml(element: ElementTree.Element, params: ConformancePar
         return greater_operation(ops[0], ops[1])
     else:
         raise ConformanceException(f'Unexpected conformance tag with children {element}')
+
+
+def parse_device_type_callable_from_xml(element: ElementTree.Element) -> Callable:
+    ''' Only allows basic, or wrappers over things that degrade to basic.'''
+    if not list(element):
+        try:
+            return parse_basic_callable_from_xml(element)
+        # For device types ONLY, there are conformances called "attributes" that are essentially just placeholders for conditions in the device library.
+        # For example, temperature controlled cabinet has conditions called "heating" and "cooling". The cluster conditions are dependent on them, but they're not
+        # actually exposed anywhere ON the device other than through the presence of the cluster. So for now, treat any attribute conditions that are cluster conditions
+        # as just optional, because it's optional to implement any device type feature.
+        # Device types also have some marked as "condition" that are similarly optional
+        except ConformanceException:
+            if element.tag == ATTRIBUTE_TAG or element.tag == CONDITION_TAG or element.tag == FEATURE_TAG:
+                return device_feature(element.attrib['name'])
+            raise
+
+    ops = []
+    for sub in element:
+        ops.append(parse_device_type_callable_from_xml(sub))
+
+    return parse_wrapper_callable_from_xml(element, ops)
+
+
+def parse_callable_from_xml(element: ElementTree.Element, params: ConformanceParseParameters) -> Callable:
+    if not list(element):
+        try:
+            return parse_basic_callable_from_xml(element)
+        except ConformanceException:
+            pass
```

Comment:
Please comment on why it's OK to `pass` here without any logs or warning.

Reply:
added comment in next commit.